### PR TITLE
Eth65 tx fetcher

### PR DESCRIPTION
**Pull request by marcindsobczak**

### CODE REVIEW

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs [hunk 1]
```diff
@@ -66,11 +68,43 @@ GetPooledTransactionsMessage getPooledTxMsg
...
+                        discoveredTxHashes.Add(hash);
+                }
+            }
+
+            int discoveredTxHashesCount = discoveredTxHashes.Count;
+
+            if (discoveredTxHashesCount != 0)
+            {
+                Send(new GetPooledTransactionsMessage(discoveredTxHashes.ToArray()));
+                Metrics.Eth65GetPooledTransactionsSent++;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs:100]: Eth65GetPooledTransactionsSent ->Eth65GetPooledTransactionsRequested ?

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs [hunk 2]
```diff
@@ -66,11 +68,43 @@ GetPooledTransactionsMessage getPooledTxMsg
...
+                    Handle(newPooledTxMsg);
                     break;
             }
         }
 
+        private void Handle(NewPooledTransactionHashesMessage msg)
+        {
+            Metrics.Eth65NewPooledTransactionHashesReceived++;
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            List<Keccak> discoveredTxHashes = new();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs:83]: Shall we use flood controller? Or not expecting issues here? If we are getting tx's that aare already pending

**Review by LukaszRozmej — APPROVED** [review]
In order to better support eht65 we might want to change how SendNewTransaction works. Instead of sending 1 tx hash, we should send by some interval batch of hashes in NewPooledTransactionHashesMessage. This is out of scope of this PR, please add ticket for it.

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs (hunk 2 above)
Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs:83]: We are receiving every hash many times and it's totally fine - sending node doesn't know if we already know this tx hash. Full txs we are receiving only on request, receiving many tx hashes isn't a problem

**Review by tkstanczak — CHANGES_REQUESTED** [review]
tests missing

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs [hunk 3]
```diff
@@ -66,11 +68,43 @@ GetPooledTransactionsMessage getPooledTxMsg
...
+                    if (_txPool.TryAddToPendingHashes(hash))
+                        discoveredTxHashes.Add(hash);
+                }
+            }
+
+            int discoveredTxHashesCount = discoveredTxHashes.Count;
+
+            if (discoveredTxHashesCount != 0)
+            {
+                Send(new GetPooledTransactionsMessage(discoveredTxHashes.ToArray()));
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs:99]: extract to a separate class that has only one responsibility -> requesting mising transactions, cover that class with unit tests, hide behind interface

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs (hunk 2 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs:83]: flood controller not needed, was only needed for eth62 geths

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs (hunk 3 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs:99]: we have added a responsibility of transaction fetching to the handler which is undesired

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs (hunk 2 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs:83]: we can still be flooded with hashes?

**Pull request merged**